week0003/truongngodang/java/Week0003.java

```diff
@@ -0,0 +1,154 @@
+import java.util.*;
+
+public class Week0003 {
+
+    private static final String ADD_NEW_CLIENT = "A";
+    private static final String DEPOSIT_MONEY = "D";
+    private static final String WITHDRAW_MONEY = "W";
+
+    public static void main(String[] args) {
+
+        Scanner scan = new Scanner(System.in);
+        HashMap<String, Client> clients = new HashMap<String, Client>();
+        ArrayList<String> output = new ArrayList<String>();
+
+        for (int i = 0; scan.hasNext(); i++) {
+            processRequest(new ProcessStringRequest(scan.nextLine()), clients, output);
+        }
+        scan.close();
+
+        output.forEach(System.out::println);
+    }
+
+    static void processRequest(ProcessStringRequest processStringRequest,
+                               HashMap<String, Client> clients,
+                               ArrayList<String> output) {
+
+            switch (processStringRequest.getDefineRequest()) {
+                case ADD_NEW_CLIENT: {
+                    if (clients.get(processStringRequest.getName()) == null
+                        && processStringRequest.getAmount() >= 0) {
+
+                        clients.put(processStringRequest.getName(),
+                                    new Client(processStringRequest.getName(),
+                                    processStringRequest.getAmount()));
+                        output.add("True");
+
+                    }
+                    else {
+                        output.add("False");
+                    }
+                    break;
+                }
+                case DEPOSIT_MONEY: {
+                    if (clients.get(processStringRequest.getName()) != null
+                        && processStringRequest.getAmount() > 0) {
+
+                        clients.get(processStringRequest.getName()).deposit(processStringRequest.getAmount());
+                        output.add("True");
+
+                    }
+                    else {
+                        output.add("False");
+                    }
+                    break;
+                }
+                case WITHDRAW_MONEY: {
+                    if (clients.get(processStringRequest.getName()) != null
+                        && clients.get(processStringRequest.getName()).getAmount() > processStringRequest.getAmount()) {
```

Comment:
`processStringRequest.getAmount()` must be positive! What if you withdraw a negative amount?